src/dash-table/components/FilterFactory.tsx

```diff
@@ -48,14 +48,14 @@ export default class FilterFactory {
 
     }
 
-    private onChange = (columnId: ColumnId, setFilter: SetFilter, ev: any) => {
-        Logger.debug('Filter -- onChange', columnId, ev.target.value && ev.target.value.trim());
+    private onChange = (column: IVisibleColumn, setFilter: SetFilter, ev: any) => {
+        Logger.debug('Filter -- onChange', column.id, ev.target.value && ev.target.value.trim());
 
         const value = ev.target.value.trim();
-        const safeColumnId = columnId.toString();
+        const safeColumnId = column.id.toString();
 
         if (value && value.length) {
-            this.ops.set(safeColumnId, new SingleColumnSyntaxTree(safeColumnId, value));
+            this.ops.set(safeColumnId, new SingleColumnSyntaxTree(value, column));
```

Comment:
Now need the id and type information in the syntax tree in order to (1) apply the default field, (2) apply the right relational operator